062-support-tiered-storage.md

```diff
@@ -0,0 +1,126 @@
+#  Support tiered storage natively in Strimzi
+
+## Current situation
+
+Kafka introduced tiered storage feature since 3.6. The feature is in early stage and it’s nice to support tiered storage configuration natively via Strimzi.
+
+## Motivation
+
+Supporting tiered storage via Strimzi allow Strimzi users to enable tiered storage natively with Strimzi config. By allowing to config custom configuration for `RemoteStorageManager`, users are able to adopt tiered storage feature on Strimzi easily.
+
+For simplicity, the proposal doesn't add support to customize `RemoteLogMetadataManager`. If tiered storage is enabled, the default implementation class `TopicBasedRemoteLogMetadataManager` in Kafka will be used.
+
+This proposal focuses on tiered storage support with a custom plugin brought in by the user. Other proposals in the future might add some built-in support for selected storage implementations such as S3. It should also be noted that, it’s on the end-user to ensure that their tiered storage setup works with Strimzi; the operator itself is strictly responsible for pushing down the necessary config to the broker pods.
+
+## Proposal
+
+Add a `spec.kafka.tieredStorage` field in the `Kafka` CR, which would support configuration for all tiered storage related configuration.
+
+The `tieredStorage` config contains configuration for some selected configuration of tiered storage feature. Most importantly, the configuration for `RemoteStorageManager` is needed.
+
+```
+kafka:
+  tieredStorage:
+    type: customRemoteStorageManager
+    className:
+    classPath:
+    config:
+    # A map with String keys and String values. Keys will be automatically prefixed with `rsm.config.`.
+
+```
+
+### RemoteStorageManager (RSM)
+
+The RSM config only supports custom type. The config allows users to specify the RSM class information.
+
+Because the `RemoteStorageManager` dependency is provided by Kafka plugin, the implementation could vary case by case. It’s important to allow user to pass in additional configuration values within the supplied prefix range. The custom configuration can be supplied via  `spec.kafka.config` with the predefined prefix.
```

Comment:
```suggestion
Because the `RemoteStorageManager` dependency is provided by a Kafka plugin, the implementation could vary case by case. It’s important to allow users to pass in additional configuration values within the supplied prefix range. The custom configuration can be supplied via  `spec.kafka.config` with the predefined prefix.
```